odbc_fdw.c

```diff
@@ -1367,3 +1576,209 @@ odbcReScanForeignScan(ForeignScanState *node)
+	{
+		/* Reflect one or more foreign tables */
+		if (options.table)
+		{
+			tables = lappend(tables, (void*)options.table);
+		}
+		else if (stmt->list_type == FDW_IMPORT_SCHEMA_ALL || stmt->list_type == FDW_IMPORT_SCHEMA_EXCEPT)
+		{
+			// TODO: Use SQLTables to obtain remote table names -> tables
+		    if (stmt->list_type == FDW_IMPORT_SCHEMA_EXCEPT)
+			{
+				// TODO: remove tables in stmt->table_list from tables
+			}
+		}
+		else if (stmt->list_type == FDW_IMPORT_SCHEMA_LIMIT_TO)
+		{
+			tables = stmt->table_list;
+		}
+		else
+		{
+			elog(ERROR,"UNKNOWN");
+		}
+        foreach(tables_cell, tables)
+		{
+			char *table_name = (char*)lfirst(tables_cell);
+
+			const char* schema_name = options.schema;
+
+			if (schema_name == NULL)
+			{
+				/* TODO: this is just a MySQL convenience; should remove it? */
+				schema_name = options.database;
+			}
+
+			odbc_connection(&options, &env, &dbc);
+
+			/* Allocate a statement handle */
+			SQLAllocHandle(SQL_HANDLE_STMT, dbc, &columns_stmt);
+
+			ret = SQLColumns(
+				 columns_stmt,
+				 NULL, 0,
+				 (SQLCHAR*)schema_name, strlen(schema_name),
+				 (SQLCHAR*)table_name,  strlen(table_name),
+				 NULL, 0
+			);
+			check_return(ret, "Obtaining ODBC columns", columns_stmt, SQL_HANDLE_STMT);
+
+            i = 0;
+			initStringInfo(&col_str);
+			ColumnName = (SQLCHAR *) palloc(sizeof(SQLCHAR) * 255);
+			while (SQL_SUCCESS == ret)
+			{
+				ret = SQLFetch(columns_stmt);
+				if (SQL_SUCCESS == ret)
+				{
+					if (++i > 1)
+					{
+						appendStringInfo(&col_str, ", ");
+					}
+					ret = SQLGetData(columns_stmt, 4, SQL_C_CHAR, ColumnName, 255, &indicator);
+					check_return(ret, "Reading column name", columns_stmt, SQL_HANDLE_STMT);
+					ret = SQLGetData(columns_stmt, 5, SQL_C_SSHORT, &DataType, 255, &indicator);
+					check_return(ret, "Reading column type", columns_stmt, SQL_HANDLE_STMT);
+					ret = SQLGetData(columns_stmt, 7, SQL_C_SLONG, &ColumnSize, 0, &indicator);
+					check_return(ret, "Reading column size", columns_stmt, SQL_HANDLE_STMT);
+					ret = SQLGetData(columns_stmt, 9, SQL_C_SSHORT, &DecimalDigits, 0, &indicator);
+					check_return(ret, "Reading column decimals", columns_stmt, SQL_HANDLE_STMT);
+					ret = SQLGetData(columns_stmt, 11, SQL_C_SSHORT, &Nullable, 0, &indicator);
+					check_return(ret, "Reading column nullable", columns_stmt, SQL_HANDLE_STMT);
+					sql_data_type(DataType, ColumnSize, DecimalDigits, Nullable, &sql_type);
+					appendStringInfo(&col_str, "\"%s\" %s", ColumnName, (char *) sql_type.data);
+				}
+			}
+			table_columns = lappend(table_columns, (void*)col_str.data);
+		}
+	}
+
+	/* Generate create statements */
+	table_columns_cell = list_head(table_columns);
+	foreach(tables_cell, tables)
+	{
+		// temporarily define vars here...
+		char *table_name = (char*)lfirst(tables_cell);
+		char *columns    = (char*)lfirst(table_columns_cell);
+		table_columns_cell = lnext(table_columns_cell);
+        StringInfoData create_statement;
```

Comment:
Here in github tabs are aligned to 8-column boundaries (instead of 4 as is standard for PG code), so this shows a case where I missed to use tabulators, and wasn't evident in my editor.